### DISCUSSION TITLE
Support of custom field on the manifest

### DISCUSSION THREAD

**Discussion by hatmaalla** [rauc/rauc#1025] [General]
Hi,
I am using rauc 1.7 in my project with **streaming capability.**
In this project, I need to accept or not the firmware update process based on some additional values to be added to the manifest.
I couldn't find in the RAUC documentation any description for this, so I'm asking if this feature is supported or not?
If not, is there a workaround to add some custom value on the manifest?

**Comment by ejoerns** [comment]
@hatmaalla We have the `version`, `description`, and `build` field in the manifest that could be used for this purpose.

There are also concrete plans to add sections for custom metadata that will not be interpreted by RAUC in any way (but could by custom logic). The pre-work for this was made in https://github.com/rauc/rauc/pull/946

**Reply by jluebbe** [reply]
PR https://github.com/rauc/rauc/pull/1026 continues the implementation of the metadata support started in #946. As stated in the PR, this needs more changes to make it accessible in hooks/handler and to the user.

**Reply by hatmaalla** [reply]
Will the value of the `meta.<label>` sections be visible from the hook using an enivronement variable like `version RAUC_MF_VERSION` or something other?

**Reply by jluebbe** [reply]
Yes, the intention is to make them available as environment variables in hooks/handlers and also in the `rauc info` output. We have some other RAUC related tasks on our list before that, though, so if you'd like to make a PR, though would be great.

**Reply by hatmaalla** [reply]
Great, can you give me an idea of when this capability will be available?

**Comment by jluebbe** [comment]
You can also also add custom files to the bundle which are then accessible to hooks.

**Reply by hatmaalla** [reply]
Thanks for your quick response, this seems like a good solution until custom metadata is officially supported. 
How do I add my custom file to the package, do I need to specify something in the manifest file?

**Reply by jluebbe** [reply]
You just add them to the the input directory passed to `rauc bundle` (next to the manifest and the image files). It doesn't need to be added to the manifest.
